File: modules/coupon-domain/src/main/java/com/huythanh0x/udemycoupons/model/coupon/CouponJsonData.java
package com.huythanh0x.udemycoupons.model.coupon;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class CouponJsonData {
    private Float price;
    private Instant expiredDate;
    private String previewImage;
    private String previewVideo;
    private int usesRemaining;
}

File: modules/coupon-domain/src/main/kotlin/com/thanh0x/coursedeal/model/coupon/CouponJsonData.kt
package com.thanh0x.coursedeal.model.coupon

import java.time.Instant

data class CouponJsonData(
    var price: Float? = null,
    var expiredDate: Instant? = null,
    var previewImage: String? = null,
    var previewVideo: String? = null,
    var usesRemaining: Int = 0
)
